test/boost/mvcc_test.cc

```diff
@@ -463,7 +463,7 @@ SEASTAR_TEST_CASE(test_apply_to_incomplete_respects_continuity) {
 
                 // After applying to_apply the continuity can be more narrow due to compaction with tombstones
                 // present in to_apply.
-                auto continuity_after = sq.get_continuity(*s);
+                auto continuity_after = e.entry().squashed_continuity(*s);
```

Comment:
In the explanation you write:

> But squashed() uses apply_monotonically(),
> so in some circumstances the squashed result can have a smaller continuity
> set than the original entry, which causes the test to fail.

But the test checks:

```
if (!continuity_after.contained_in(e_continuity)) {
```

So it already allows the result to have "smaller continuity set than the original entry". 


In the issue, the problem stated is:

> With the problematic seed, the initial entry contains a row_entry (discontinuous on both sides) with nothing but a row_tombstone with timestamp older than the partition tombstone. When the "before apply" continuity is calculated, this useless entry is ignored and doesn't go into the continuity set.
> 
> The mutation applied in apply_to_incomplete() covers the useless row_entry with a range tombstone. Since redundant range tombstones aren't cleared, maybe_drop doesn't see anymore that the entry is useless, so it isn't purged during the squash when calculating continuity "after apply", and the test fails because apparently continuity grew after apply_to_incomplete().
>

Reply:
> But squashed() uses apply_monotonically(), so in some circumstances the squashed result can have a smaller continuity set than the original entry

In this sentence, by "original entry" I meant the argument of `squashed()`, not anything related to the test.
I updated the description to hopefully remove the confusion.

Reply:
Ah, and the problematic "original entry" is the "before" entry? The "after" entry is not affected, hence it has wider continuity than "before" entry.

Reply:
> Ah, and the problematic "original entry" is the "before" entry?

Yes. In the failing test runs, the "before" entry has an empty row discontinuous on both sides. This row gets dropped by `squashed()`. In the "after" entry, this empty row has been "painted" with a range tombstone, so `maybe_drop()` is no longer allowed to drop it, so `squashed()` doesn't drop it. And in the end, "after" has (apparently) bigger continuity than the "before".